fe/fe-core/src/main/java/org/apache/doris/common/util/FetchRemoteTabletSchemaUtil.java

```diff
@@ -0,0 +1,323 @@
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.common.util;
+
+import org.apache.doris.catalog.AggregateType;
+import org.apache.doris.catalog.ArrayType;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.Env;
+import org.apache.doris.catalog.MapType;
+import org.apache.doris.catalog.Replica;
+import org.apache.doris.catalog.StructType;
+import org.apache.doris.catalog.Tablet;
+import org.apache.doris.catalog.Type;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.proto.InternalService.PFetchRemoteSchemaRequest;
+import org.apache.doris.proto.InternalService.PFetchRemoteSchemaResponse;
+import org.apache.doris.proto.InternalService.PTabletsLocation;
+import org.apache.doris.proto.OlapFile.ColumnPB;
+import org.apache.doris.proto.OlapFile.TabletSchemaPB;
+import org.apache.doris.rpc.BackendServiceProxy;
+import org.apache.doris.rpc.RpcException;
+import org.apache.doris.system.Backend;
+import org.apache.doris.thrift.TStatusCode;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+
+public class FetchRemoteTabletSchemaUtil {
+    private static final Logger LOG = LogManager.getLogger(FetchRemoteTabletSchemaUtil.class);
+
+    private List<Tablet> tablets;
+    private List<Column> columns;
+
+    public FetchRemoteTabletSchemaUtil(List<Tablet> tablets) {
+        this.tablets = tablets;
+        this.columns = Lists.newArrayList();
+    }
+
+    public List<Column> fetch() {
+        // find be
+        Preconditions.checkNotNull(tablets);
+        Map<Long, Set<Long>> beIdToTabletId = Maps.newHashMap();
+        for (Tablet tablet : tablets) {
+            for (Replica replica : tablet.getReplicas()) {
+                Set<Long> tabletIds = beIdToTabletId.computeIfAbsent(
+                                    replica.getBackendId(), k -> Sets.newHashSet());
+                tabletIds.add(tablet.getId());
+            }
+        }
+
+        // build PTabletsLocation
+        List<PTabletsLocation> locations = Lists.newArrayList();
+        List<Backend> coordinatorBackend = Lists.newArrayList();
+        for (Map.Entry<Long, Set<Long>> entry : beIdToTabletId.entrySet()) {
+            Long backendId = entry.getKey();
+            Set<Long> tabletIds = entry.getValue();
+            Backend backend = Env.getCurrentEnv().getCurrentSystemInfo().getBackend(backendId);
+            if (coordinatorBackend.size() < 2) {
```

Comment:
`< 2` is meanling leass, make it more clear.Or add comment

Reply:
done

Reply:
I have added some comments